packages/@aws-cdk/core/lib/private/resolve.ts

```diff
@@ -217,11 +217,12 @@ export function resolve(obj: any, options: IResolveOptions): any {
   // objects - deep-resolve all values
   //
 
-  // Must not be a Construct at this point, otherwise you probably made a typo
+  // Must not be a non-pure data object at this point, otherwise you probably made a typo
```

Comment:
Can we get rid of the double negation? 🤓

```suggestion
  // Must be a pure data object at this point, otherwise you probably made a typo
```